server/docker-base/template.csproj

```diff
@@ -0,0 +1,34 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net9.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <PublishAot>true</PublishAot>
+    <NativeLib>static</NativeLib>
+    <OutputType>Library</OutputType>
+    <PublishTrimmed>true</PublishTrimmed>
+    <IlcDehydrate>false</IlcDehydrate>
+    <Optimize>true</Optimize>
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+
+    <!-- Workaround for restoring iOS runtime packs for NativeAOT -->
+    <PublishAotUsingRuntimePack>true</PublishAotUsingRuntimePack>
+    <HybridGlobalization>false</HybridGlobalization>
+
+    <TargetName>$(MSBuildProjectName)</TargetName>
+    <TargetName Condition="'$(OS)' != 'Windows_NT'">lib$(TargetName)</TargetName>
+    <StackTraceSupport>true</StackTraceSupport>
+    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+    <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
+    <IntermediateOutputPath>{{BUILDDIR_CS}}</IntermediateOutputPath>
+    <OutputPath>{{BUILDDIR_CS}}</OutputPath>
+  </PropertyGroup>
+  <ItemGroup>
+    <!-- <PackageReference Include="Microsoft.DotNet.ILCompiler" Version="9.0.0" /> -->
+
+    {{#PINVOKE}}
+    <DirectPInvoke Include="{{{.}}}" />
```

Comment:
We list each engine library here, to allow the C# files to map to these libraries.